lightning/src/ln/channel.rs

```diff
@@ -1495,7 +1495,148 @@ pub(super) struct ChannelContext<SP: Deref> where SP::Target: SignerProvider {
+
+		assert!(!context.channel_state.is_monitor_update_in_progress()); // We have not had any monitor(s) yet to fail update!
+		if context.is_batch_funding() {
+			context.channel_state = ChannelState::AwaitingChannelReady(AwaitingChannelReadyFlags::WAITING_FOR_BATCH);
+		} else {
+			context.channel_state = ChannelState::AwaitingChannelReady(AwaitingChannelReadyFlags::new());
+		}
+		if context.holder_commitment_point.advance(&context.holder_signer, &context.secp_ctx, logger).is_err() {
+			// We only fail to advance our commitment point/number if we're currently
+			// waiting for our signer to unblock and provide a commitment point.
+			// We cannot send accept_channel/open_channel before this has occurred, so if we
+			// err here by the time we receive funding_created/funding_signed, something has gone wrong.
+			debug_assert!(false, "We should be ready to advance our commitment point by the time we receive {}", self.received_msg());
+			return Err(ChannelError::close("Failed to advance holder commitment point".to_owned()));
+		}
+
+		let funding_redeemscript = context.get_funding_redeemscript();
+		let funding_txo = context.get_funding_txo().unwrap();
+		let funding_txo_script = funding_redeemscript.to_p2wsh();
+		let obscure_factor = get_commitment_transaction_number_obscure_factor(&context.get_holder_pubkeys().payment_point, &context.get_counterparty_pubkeys().payment_point, context.is_outbound());
+		let shutdown_script = context.shutdown_scriptpubkey.clone().map(|script| script.into_inner());
+		let mut monitor_signer = signer_provider.derive_channel_signer(context.channel_value_satoshis, context.channel_keys_id);
+		monitor_signer.provide_channel_parameters(&context.channel_transaction_parameters);
+		let channel_monitor = ChannelMonitor::new(context.secp_ctx.clone(), monitor_signer,
+		                                          shutdown_script, context.get_holder_selected_contest_delay(),
+		                                          &context.destination_script, (funding_txo, funding_txo_script),
+		                                          &context.channel_transaction_parameters, context.is_outbound(),
+		                                          funding_redeemscript.clone(), context.channel_value_satoshis,
+		                                          obscure_factor,
+		                                          holder_commitment_tx, best_block, context.counterparty_node_id, context.channel_id());
+		channel_monitor.provide_initial_counterparty_commitment_tx(
+			counterparty_txid, Vec::new(),
+			counterparty_commitment_number,
+			context.counterparty_cur_commitment_point.unwrap(),
+			counterparty_initial_commitment_tx.feerate_per_kw(),
+			counterparty_initial_commitment_tx.to_broadcaster_value_sat(),
+			counterparty_initial_commitment_tx.to_countersignatory_value_sat(),
+			logger);
+
+		Ok(channel_monitor)
+	}
+}
+
+impl<SP: Deref> InitialRemoteCommitmentReceiver<SP> for OutboundV1Channel<SP> where SP::Target: SignerProvider {
```

Comment:
I think it would make sense to merge `OutboundV1Channel` and `InboundV1Channel` into one struct with one additional `is_outbound` field, the differences are minimal. If we are DRYing :)

Reply:
It still provides type safety on methods that are not the same, like `funding_signed` and `funding_created` and a few other aspects.